File: build-manifest.js
#!/usr/bin/env node

/**
 * Build script to generate games-manifest.json
 * Run this script whenever you add new games to the games/ folder
 * Usage: node build-manifest.js
 */

const fs = require('fs');
const path = require('path');

const GAMES_DIR = path.join(__dirname, 'games');
const OUTPUT_FILE = path.join(__dirname, 'games-manifest.json');
const EXCLUDE_FILES = ['README.md', 'template.html', '.gitkeep'];

function scanGamesDirectory() {
    try {
        const files = fs.readdirSync(GAMES_DIR);
        const gameFiles = files.filter(file => {
            return file.endsWith('.html') && !EXCLUDE_FILES.includes(file);
        });
        
        console.log(`Found ${gameFiles.length} game(s):`);
        gameFiles.forEach(file => console.log(`  - ${file}`));
        
        return gameFiles;
    } catch (error) {
        console.error('Error scanning games directory:', error);
        return [];
    }
}

function generateManifest() {
    const games = scanGamesDirectory();
    
    const manifest = {
        generated: new Date().toISOString(),
        count: games.length,
        games: games
    };
    
    fs.writeFileSync(OUTPUT_FILE, JSON.stringify(manifest, null, 2));
    console.log(`\n✅ Manifest generated: ${OUTPUT_FILE}`);
    console.log(`Total games: ${games.length}`);
}

// Run the script
generateManifest();

File: build-manifest.py
#!/usr/bin/env python3

"""
Build script to generate games-manifest.json
Run this script whenever you add new games to the games/ folder
Usage: python3 build-manifest.py
"""

import os
import json
from datetime import datetime

GAMES_DIR = 'games'
OUTPUT_FILE = 'games-manifest.json'
EXCLUDE_FILES = ['README.md', 'template.html', '.gitkeep']

def scan_games_directory():
    """Scan the games directory for HTML files"""
    try:
        files = os.listdir(GAMES_DIR)
        game_files = [
            f for f in files 
            if f.endswith('.html') and f not in EXCLUDE_FILES
        ]
        
        print(f"Found {len(game_files)} game(s):")
        for file in game_files:
            print(f"  - {file}")
        
        return game_files
    except Exception as e:
        print(f"Error scanning games directory: {e}")
        return []

def generate_manifest():
    """Generate the games manifest JSON file"""
    games = scan_games_directory()
    
    manifest = {
        'generated': datetime.utcnow().isoformat() + 'Z',
        'count': len(games),
        'games': sorted(games)
    }
    
    with open(OUTPUT_FILE, 'w') as f:
        json.dump(manifest, f, indent=2)
    
    print(f"\n✅ Manifest generated: {OUTPUT_FILE}")
    print(f"Total games: {len(games)}")

if __name__ == '__main__':
    generate_manifest()